lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
-    it('executes onClose method with a cancel event when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onClose: mockCallBack });
+    it('executes onClose method with a cancel event when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
+      quickRender({ onClose: mockCallBack });
       expect(mockCallBack).not.toHaveBeenCalled();
-      component.simulate('keyDown', createEvent());
+
+      await user.keyboard('[Escape]');
+
       expect(mockCallBack).toHaveBeenCalledTimes(1);
       expect(mockCallBack.mock.calls[0][0].type).toBe('cancel');
     });
 
-    it('executes onCancel method with a cancel event when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onCancel: mockCallBack });
+    it('executes onCancel method with a cancel event when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
+      quickRender({ onCancel: mockCallBack });
       expect(mockCallBack).not.toHaveBeenCalled();
-      component.simulate('keyDown', createEvent());
+
+      await user.keyboard('[Escape]');
+
       expect(mockCallBack).toHaveBeenCalledTimes(1);
       expect(mockCallBack.mock.calls[0][0].type).toBe('cancel');
     });
 
-    it('executes onClose method ONLY when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onClose: mockCallBack });
+    it('executes onClose method ONLY when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
-      component.simulate('keyDown', createEvent('Tab'));
-      component.simulate('keyDown', createEvent('Enter'));
-      component.simulate('keyDown', createEvent('q'));
-      component.simulate('keyDown', createEvent('Q'));
-      expect(mockCallBack).not.toHaveBeenCalled();
-    });
+      quickRender({ onClose: mockCallBack });
 
-    it('executes onCancel method ONLY when pressing ESC key', function () {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onCancel: mockCallBack });
+      await user.keyboard('[Tab]');
+      await user.keyboard('[Enter]');
+      await user.keyboard('q');
+      await user.keyboard('Q');
 
-      component.simulate('keyDown', createEvent('Tab'));
-      component.simulate('keyDown', createEvent('Enter'));
-      component.simulate('keyDown', createEvent('q'));
-      component.simulate('keyDown', createEvent('Q'));
       expect(mockCallBack).not.toHaveBeenCalled();
     });
 
-    it('calls stopPropagation and stopImmediatePropagation on Escape keydowns', function() {
-      const component = getModal({ onClose: jest.fn() }),
-          escEvent = createEvent(),
-          otherEvent = createEvent('q');
+    it('executes onCancel method ONLY when pressing ESC key', async function () {
```

Comment:
does onCancel get called when you close the alert by clicking the close button in the modal footer?

Reply:
> does onCancel get called when you close the alert by clicking the close button in the modal footer? And if that is true, should there be a test proving the modal doesn't close when clicking on the dialog element/modal backdrop?

`onCancel` doesn't get called when a button in the modal footer is clicked.

Reply:
> `onCancel` doesn't get called when a button in the modal footer is clicked.

Thanks for clarifying!